README.md

```diff
@@ -1,4 +1,102 @@
 # KindedJ
 A collection of shared interfaces for evidence-based Higher Kinded Types in the JVM
 
-Gitter: https://gitter.im/KindedJ/Lobby
+[![Gitter chat](https://badges.gitter.im/KindedJ/KindedJ.png)](https://gitter.im/KindedJ/Lobby)
+
+## Projects that support KindedJ
+
+*Add yourself here alphabetically by [sending a Pull Request](https://github.com/KindedJ/KindedJ/compare)*
+
+| | Name & Organisation | Language | Since version | Description |
+|:---:| --- |:---:| ---:| --- |
+<img src="https://raw.githubusercontent.com/kategory/kategory-art/master/kategory-brand-light.png" width="48">|[Kategory](https://github.com/kategory/kategory)|Kotlin|0.3.6|Functional Data Types & Abstractions for Kotlin|
+
+## Purpose
+
+The objective of KindedJ is to provide a common set of interfaces that enable cross-library and cross-language support for all projects using evidence-based Higher Kinded Types emulation in the JVM.
+
+A Higher Kinded Type represents a value that wraps another value. They can be reasoned about as type constructors, where arity represents the number of generic parameters.
```

Comment:
I would avoid talking about "value" here as it is a separate concern. What about something like:
"
Higher Kinded Types is feature of a type system which main purpose is to allow *type constructor polymorphism*, that is abstracting over type constructors.
So what is a type constructor? It is a function at the type level that take type(s) as argument(s) and return a type as a result.
Eg. we can say that `List` is a type constructor, it as one type parameter: it need one type argument to *produce* a concrete type. `List` applied with the type argument `String` gives you the concrete type `List<String>`

Abstracting over type constructor means declaring methods that are generic on the type constructor. The archetype of such a method is the `map` method. in Java this method exists in `java.util.Optional`, `java.util.stream.Stream` and `java.util.concurrent.CompletableFuture` (under the name `thenApply`).
Now if you want to write a reusable piece of code that can work for every types that implement `map`, without HKT you actually need to duplicate that code (and the possibility of errors) for every type you wish to support. With HKT, you only write it once.
"

Reply:
Agree with @jbgi on this. Probably a concise version of the above would make for a good description. My only concern is whether the definition with type constructors will mean anything to a lot of people. I guess the better question is: who is this README designed for? Just library devs like us who want a common interface for HKTs or for people who want to do HKT abstraction (so functional JVM programmers)?

The brief example of Java standard lib APIs as @jbgi mentioned above is very effective and motivates why we need HKTs in the first place - to reduce boilerplate.